keps/sig-node/127-user-namespaces/README.md

```diff
@@ -200,6 +201,65 @@ If true or not present, uses the host user namespace (as today)
 If false, a new userns is created for the pod.
 By default it is set to `true`.
 
+### CRI changes
+
+The following messages will be added:
+
+```
+// A IDMapping describes a ID mapping for the user namespace used for the
+// pod.
+message IDMapping {
+    // host_id is the id on the host.
+    uint32 host_id = 1;
+    // container_id is the id in the container.
+    uint32 container_id = 2;
+    // length is the size of the range to map.
+    uint32 length = 3;
+}
+
+// A UserNamespace describes the intended user namespace configuration.
+message UserNamespace {
+    // User namespace for this sandbox.
+    // Note: It currently supports only POD and NODE.
+    NamespaceMode mode = 1;
+
+    // uids specifies the UID mappings for the user namespace.
+    repeated IDMapping uids = 2;
+
+    // gids specifies the GID mappings for the user namespace.
+    repeated IDMapping gids = 3;
+}
+```
+
+The existing message `NamespaceOption` will have a `user` field added.
+The complete `NamespaceOption` message with the new field is shown here:
+
+```
+// NamespaceOption provides options for Linux namespaces.
+message NamespaceOption {
```

Comment:
should these options be 'optional' so some of them do not have to be set?

Reply:
This is protocol buffers 3, that is what the CRI api uses, all fields are optional. The generated code creates this as a pointer, so we can easily see nil (or don't assign it on the sending part) when unsued, etc.

Reply:
We have also tested this in as expected in the PoC implementations we did :)